Linter to avoid csv merge conflicts

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #111163


This PR addresses the persistent issue of merge conflicts in the benchmarks/dynamo/ci_expected_accuracy/ directory, specifically those arising from frequently updated CSV files. Based on @malfet  suggestion, the solution implemented adds three spaces between each line in the CSV files. This approach has proven effective in preventing merge conflicts, as evidenced in [D50239634](https://www.internalfb.com/intern/diff/D50239634/). Regardless of these changes the extra new lines should still allow the csvs to be ingested as normal.

If you have access to the diff:
Normally, modifying a line that is later altered in the stack results in a merge conflict during restacking. With this new spacing strategy, lines that are not modified further down the stack will not trigger merge conflicts, achieving our intended outcome.

cc @albanD @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

PaliC committed:
new linter

[ghstack-poisoned]

PaliC committed:
Update on "new linter"




[ghstack-poisoned]

malfet reviewed (approved):
LGTM, if it passes CI (specifically linter and torchbench tests)

kit1980 commented on tools/linter/adapters/no_merge_conflict_csv_linter.py:
```diff
@@ -0,0 +1,132 @@
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Format files with ufmt (black + usort).",
```
Is this correct description?

kit1980 commented on tools/linter/adapters/no_merge_conflict_csv_linter.py:
```diff
@@ -0,0 +1,132 @@
+                path=filename,
+                line=None,
+                char=None,
+                code="UFMT",
```
UFMT or a different code?

huydhn replied:
I guess it should be `MERGE_CONFLICTLESS_CSV`?

kit1980 replied:
It's still UFMT?

PaliC committed:
Update on "Linter to avoid csv merge conflicts"



This PR addresses the persistent issue of merge conflicts in the benchmarks/dynamo/ci_expected_accuracy/ directory, specifically those arising from frequently updated CSV files. Based on malfet  suggestion, the solution implemented adds three spaces between each line in the CSV files. This approach has proven effective in preventing merge conflicts, as evidenced in [D50239634](https://www.internalfb.com/intern/diff/D50239634/). Regardless of these changes the extra new lines should still allow the csvs to be ingested as normal.

If you have access to the diff:
Normally, modifying a line that is later altered in the stack results in a merge conflict during restacking. With this new spacing strategy, lines that are not modified further down the stack will not trigger merge conflicts, achieving our intended outcome.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

huydhn commented on tools/linter/adapters/no_merge_conflict_csv_linter.py:
```diff
@@ -0,0 +1,132 @@
+    replacement = ""
+    with open(filename, 'r') as f:
+        lines = f.readlines()
+        for line in lines:
```
Hmm, is this going to add 3 new lines to the file every time I run lintrunner, so the file would keep growing.  I need to try this out locally :)

PaliC replied:
It does not do this, at least when I tested this. Strip should return an empty line for the new lines

kit1980 reviewed (changes requested):
Please clean-up unused code.
For example, `format_error_message` and `as_posix` doesn't look like they are used. And some other things.

PaliC committed:
Update on "Linter to avoid csv merge conflicts"



This PR addresses the persistent issue of merge conflicts in the benchmarks/dynamo/ci_expected_accuracy/ directory, specifically those arising from frequently updated CSV files. Based on malfet  suggestion, the solution implemented adds three spaces between each line in the CSV files. This approach has proven effective in preventing merge conflicts, as evidenced in [D50239634](https://www.internalfb.com/intern/diff/D50239634/). Regardless of these changes the extra new lines should still allow the csvs to be ingested as normal.

If you have access to the diff:
Normally, modifying a line that is later altered in the stack results in a merge conflict during restacking. With this new spacing strategy, lines that are not modified further down the stack will not trigger merge conflicts, achieving our intended outcome.

cc albanD voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

PaliC commented:
@pytorchbot merge

PaliC committed:
Update on "Linter to avoid csv merge conflicts"



This PR addresses the persistent issue of merge conflicts in the benchmarks/dynamo/ci_expected_accuracy/ directory, specifically those arising from frequently updated CSV files. Based on malfet  suggestion, the solution implemented adds three spaces between each line in the CSV files. This approach has proven effective in preventing merge conflicts, as evidenced in [D50239634](https://www.internalfb.com/intern/diff/D50239634/). Regardless of these changes the extra new lines should still allow the csvs to be ingested as normal.

If you have access to the diff:
Normally, modifying a line that is later altered in the stack results in a merge conflict during restacking. With this new spacing strategy, lines that are not modified further down the stack will not trigger merge conflicts, achieving our intended outcome.

cc albanD voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

PaliC commented:
@pytorchbot merge
